build-tools/xa-prep-tasks/Xamarin.Android.BuildTools.PrepTasks/CheckoutExternalGitDependency.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.IO;
+using Microsoft.Build.Framework;
+
+namespace Xamarin.Android.BuildTools.PrepTasks
+{
+	public class CheckoutExternalGitDependency : Git
+	{
+		[Required]
+		public ITaskItem ExternalGitDependency { get; set; }
+
+		protected override bool LogTaskMessages {
+			get { return false; }
+		}
+
+		string commit;
+		string owner;
+		string name;
+
+		public override bool Execute ()
+		{
+			commit = ExternalGitDependency.ItemSpec;
+			owner = ExternalGitDependency.GetMetadata ("Owner");
+			name = ExternalGitDependency.GetMetadata ("Name");
+			string destination = Path.Combine (GetWorkingDirectory (), name);
+
+			if (!Directory.Exists (destination)) {
+				Clone ();
+			}
+
+			WorkingDirectory.ItemSpec = destination;
+			Fetch ();
+			CheckoutCommit ();
+
+			return !Log.HasLoggedErrors;
+		}
+
+		void Clone ()
+		{
+			string ghToken = Environment.GetEnvironmentVariable("GH_AUTH_SECRET");
```

Comment:
Should this method also set `WorkingDirectory.ItemSpec` before the `base.Execute()`?  Additionally, should the `clone` command explicitly specify the output directory?

Reply:
`WorkingDirectory.ItemSpec` will be set by https://github.com/xamarin/xamarin-android/pull/2845/files/22c6222bc0ac1f78e98e162c64397c6809baf137#diff-d847e8261f0bc18e3ae50dd771aabea9R19, so there should be no need to update it before Cloning if needed. We could explicitly specify the output directory when cloning but I believe it will be redundant, as the repo name in the `.external` file will always match the directory name which the repo is cloned into.